test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/RuntimeTypeInfoTest.cs

```diff
@@ -0,0 +1,366 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    public class RuntimeTypeInfoTest
+    {
+        [Theory]
+        [InlineData(typeof(int))]
+        [InlineData(typeof(string))]
+        [InlineData(typeof(Tuple<,>))]
+        [InlineData(typeof(IDictionary<string, string>))]
+        [InlineData(typeof(AbstractType))]
+        [InlineData(typeof(PrivateType))]
+        public void RuntimeTypeInfo_ReturnsMetadataOfAdaptingType(Type type)
+        {
+            // Arrange
+            var typeInfo = type.GetTypeInfo();
+            var runtimeTypeInfo = new RuntimeTypeInfo(typeInfo);
+
+            // Act and Assert
+            Assert.Same(runtimeTypeInfo.TypeInfo, typeInfo);
+            Assert.Equal(runtimeTypeInfo.Name, typeInfo.Name);
+            Assert.Equal(runtimeTypeInfo.FullName, typeInfo.FullName);
+            Assert.Equal(runtimeTypeInfo.IsAbstract, typeInfo.IsAbstract);
+            Assert.Equal(runtimeTypeInfo.IsGenericType, typeInfo.IsGenericType);
+            Assert.Equal(runtimeTypeInfo.IsPublic, typeInfo.IsPublic);
+        }
+
+        [Fact]
+        public void Properties_ReturnsPublicPropertiesOfAdaptingType()
+        {
+            // Arrange
+            var typeInfo = typeof(SubType).GetTypeInfo();
+            var runtimeTypeInfo = new RuntimeTypeInfo(typeInfo);
+
+            // Act and Assert
+            Assert.Collection(runtimeTypeInfo.Properties,
```

Comment:
This asserts a particular order. Even if the test works fine on all platforms today, would anything break if the order changed in the future? If not, explicitly order `Properties` to ignore their original order.